src/proto/git_f_s.proto

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright 2020 Nick Mertin
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+syntax = "proto3";
+
+package bowler.git_f_s;
+
+option java_multiple_files = true;
+option java_package = "com.commonwealthrobotics.proto.gitfs";
+
+// Next: 4
+// The host-agnostic identifier for a project
+message ProjectSpec {
+    // The Git remote path containing the project
+    string repo_remote = 1;
+
+    // The Git commit-ish identifying the Git revision of the file
+    string revision = 2;
+
+    // A set of local changes to the project
+    repeated FilePatch patches = 3;
+}
+
+// Next: 3
+// The host-agnostic identifier for a file in a project
+message FileSpec {
+    // The project containing the file
+    ProjectSpec project = 1;
+
+    // The relative path to the file within the repository
+    string path = 2;
+}
+
+// Next: 5
+// The description of local modifications to a file within a repository
+message FilePatch {
```

Comment:
I don't want us to reinvent the "git patch wheel". We could implement patching by literally using git to create and apply patches, so what do you think about modeling patches as strings?

Reply:
Using `bytes`, to ensure proper support for binary files.